packages/vite/src/node/optimizer/resolve.ts

```diff
@@ -89,22 +90,12 @@ export function expandGlobIds(id: string, config: ResolvedConfig): string[] {
           )
 
           possibleExportPaths.push(
-            ...glob
-              .sync(exportValuePattern, {
-                cwd: pkgData.dir,
-                ignore: ['node_modules'],
-              })
+            ...globSync(exportValuePattern, {
+              cwd: pkgData.dir,
+              expandDirectories: false,
+              ignore: ['node_modules'],
+            })
               .map((filePath) => {
-                // ensure "./" prefix for inconsistent fast-glob result
-                //   glob.sync("./some-dir/**/*") -> "./some-dir/some-file"
-                //   glob.sync("./**/*")          -> "some-dir/some-file"
-                if (
-                  exportsValue.startsWith('./') &&
-                  !filePath.startsWith('./')
-                ) {
-                  filePath = './' + filePath
-                }
```

Comment:
`tinyglobby` is consistent in that both `"./some-dir/**/*"` and `"./**/*"` return `some-dir/some-file`. Removing the code here is one solution, the other one would be to *always* add leading `"./"` if the pattern starts with that. What do you think?